github4s/src/test/scala/github4s/integration/ReposSpec.scala

```diff
@@ -121,6 +121,38 @@ trait ReposSpec extends BaseIntegrationSpec {
     response.statusCode shouldBe okStatusCode
   }
 
+  "Repos >> GetContents" should "have the same contents with getBlob using fileSha" taggedAs Integration in {
+    val gh = clientResource.map { client =>
+      Github[IO](client, accessToken)
+    }
+
+    val response: GHResponse[NonEmptyList[Content]] = gh
+      .use(
+        _.repos.getContents(validRepoOwner, validRepoName, validFilePath, headers = headerUserAgent)
+      )
+      .unsafeRunSync()
+
+    val fileContentO = for {
+      res     <- response.result.toOption
+      content <- res.head.content
+    } yield content
+
+    fileContentO.isDefined shouldBe true
+
+    val blobResponse = for {
+      res <- response.result.toOption
+    } yield gh.use(
+      _.gitData.getBlob(validRepoOwner, validRepoName, res.head.sha, headers = headerUserAgent)
+    )
+
+    blobResponse.isDefined shouldBe true
+
+    val responseBlob = blobResponse.get.unsafeRunSync()
+
+    testIsRight[BlobContent](responseBlob, r => r.content shouldBe fileContentO)
+    response.statusCode shouldBe okStatusCode
```

Comment:
I think this can be simplified a bit: https://gist.github.com/BenFradet/4d5e85580ef986d7c39300d5690af867

Reply:
How about this approach?

Reply:
yup that's cool too :+1: